packages/fether-electron/src/main/index.js

```diff
@@ -3,151 +3,53 @@
-import staticPath from './utils/staticPath';
+import Pino from './app/utils/pino';
+import FetherApp from './app';
+import FetherAppOptions from './app/options';
 
-const { app, BrowserWindow, ipcMain, session } = electron;
-let mainWindow;
+const { app } = electron;
 const pino = Pino();
 
+let withTaskbar = process.env.TASKBAR !== 'false';
+
 // Disable gpu acceleration on linux
 // https://github.com/parity-js/fether/issues/85
 if (!['darwin', 'win32'].includes(process.platform)) {
   app.disableHardwareAcceleration();
 }
 
-function createWindow () {
-  pino.info(`Starting ${productName}...`);
-  mainWindow = new BrowserWindow({
-    height: 640,
-    resizable: false,
-    width: 360
-  });
-
-  // Set options for @parity/electron
-  parityElectron({
-    logger: namespace => log => Pino({ name: namespace }).info(log)
-  });
-
-  // Look if Parity is installed
-  getParityPath()
-    .catch(() =>
-      // Install parity if not present
-      fetchParity(mainWindow, {
-        onProgress: progress =>
-          // Notify the renderers on download progress
-          mainWindow.webContents.send('parity-download-progress', progress),
-        parityChannel: parity.channel
-      })
-    )
-    .then(async () => {
-      // Run parity when installed
-
-      // Don't run parity if the user ran fether with --no-run-parity
-      if (!cli.runParity) {
-        return;
-      }
-
-      if (
-        await isParityRunning({
-          wsInterface: cli.wsInterface,
-          wsPort: cli.wsPort
-        })
-      ) {
-        return;
-      }
-
-      return runParity({
-        flags: [
-          ...getRemainingArgs(cli),
-          '--light',
-          '--chain',
-          cli.chain,
-          '--ws-interface',
-          cli.wsInterface,
-          '--ws-port',
-          cli.wsPort
-        ],
-        onParityError: err => handleError(err, 'An error occured with Parity.')
-      });
-    })
-    .then(() => {
-      // Notify the renderers
-      mainWindow.webContents.send('parity-running', true);
-      global.isParityRunning = true; // Send this variable to renderes via IPC
-    })
-    .catch(handleError);
-
-  // Globals for fether-react parityStore
-  global.wsInterface = cli.wsInterface;
-  global.wsPort = cli.wsPort;
-
-  // Opens file:///path/to/build/index.html in prod mode, or whatever is
-  // passed to ELECTRON_START_URL
-  mainWindow.loadURL(
-    process.env.ELECTRON_START_URL ||
-      url.format({
-        pathname: path.join(staticPath, 'build', 'index.html'),
-        protocol: 'file:',
-        slashes: true
-      })
-  );
-
-  // Listen to messages from renderer process
-  ipcMain.on('asynchronous-message', (...args) =>
-    messages(mainWindow, ...args)
-  );
+if (process.platform === 'win32') {
+  withTaskbar = false;
+}
 
-  // Add application menu
-  addMenu(mainWindow);
+const fetherAppInstance = new FetherApp();
+const fetherAppOptionsInstance = new FetherAppOptions();
```

Comment:
Is a class a bit overkill for some options? Unless the class does something special?

Reply:
Yes its overkill